qualtran/bloqs/gf_arithmetic/gf2_multiplication.py

```diff
@@ -268,37 +268,44 @@ class GF2MultiplyByConstantMod(Bloq):
     """
 
     const: 'galois.FieldArray'
-    galois_field: 'galois.FieldArrayMeta'
```

Comment:
I'd also suggest that we change the `const` type to `SymbolicInt` instead of `FieldArray` and do a validation in `__attrs_post_init__` if needed. This is helpful because it allows us to instanstiate the bloq when `const` is symbolic and also allows us to keep the hash simple. This is also consistent with what is done in `GF2AddK`.

Reply:
done